Sources/StreamChat/Repositories/MessageRepository.swift

```diff
@@ -257,4 +257,37 @@ class MessageRepository {
             completion?()
         }
     }
+    
+    func undoMessagePinning(
+        on messageId: MessageId,
+        completion: (() -> Void)? = nil
+    ) {
+        database.write { session in
+            if let messageDTO = session.message(id: messageId) {
+                session.unpin(message: messageDTO)
+            }
+        } completion: { error in
+            if let error = error {
+                log.error("Error unpinning the message with id \(messageId): \(error)")
+            }
+            completion?()
+        }
+    }
+    
+    func undoMessageUnpinning(
+        on messageId: MessageId,
+        completion: (() -> Void)? = nil
+    ) {
+        database.write { session in
+            if let messageDTO = session.message(id: messageId) {
+                let pinning: MessagePinning = .init(expirationDate: messageDTO.pinExpires?.bridgeDate)
+                try session.pin(message: messageDTO, pinning: pinning)
```

Comment:
Does it really work like this?
If we call `session.unpin(…)` then this sets `pinExpires` to nil therefore we lose this information and calling `undoMessageUnpinning` can't recover this information anymore.

Wouldn't it be better not to have undo methods here and do everything within the `MessageUpdater`?

Reply:
Oh, you're right.

I updated the function and now pass `pinning` as an argument. Check it out 🙂